### PR TITLE
Core/static_autoscaler.go unit tests.

### PR DESCRIPTION
**Pull request by krzysztof-jastrzebski**

### CODE REVIEW

**Comment by krzysztof-jastrzebski** [comment]
Coverage improvement:
package cluster-autoscaler.core - 64.5% -> 74.2
file core/static_autoscaler.go - 3.2% - > 62.3%

File: cluster-autoscaler/core/static_autoscaler_test.go [hunk 1]
```diff
@@ -0,0 +1,238 @@
+/*
+Copyright 2016 The Kubernetes Authors.
```
Comment (mwielgus) [inline cluster-autoscaler/core/static_autoscaler_test.go]: Date.

File: cluster-autoscaler/core/static_autoscaler_test.go [hunk 2]
```diff
@@ -0,0 +1,238 @@
...
+	podDisruptionBudgetListerMock.On("List").Return([]*policyv1.PodDisruptionBudget{}, nil).Once()
+
+	provider.AddNode("ng1", n2)
+
+	err = autoscaler.RunOnce(time.Now())
+	assert.NoError(t, err)
+	mock.AssertExpectationsForObjects(t, readyNodeListerMock, allNodeListerMock, scheduledPodMock, unschedulablePodMock,
+		podDisruptionBudgetListerMock, daemonSetListerMock, onScaleUpMock, onScaleDownMock)
+
+	// Scale down.
```
Comment (mwielgus) [inline cluster-autoscaler/core/static_autoscaler_test.go]: Please add a comment (or snippet) why scale down is possible just after scale up.

Comment (krzysztof-jastrzebski) [reply cluster-autoscaler/core/static_autoscaler_test.go]: Done

File: cluster-autoscaler/core/static_autoscaler_test.go (hunk 1 above)
Comment (krzysztof-jastrzebski) [reply cluster-autoscaler/core/static_autoscaler_test.go]: Done

File: cluster-autoscaler/core/static_autoscaler_test.go (hunk 2 above)
Comment (krzysztof-jastrzebski) [reply cluster-autoscaler/core/static_autoscaler_test.go]: Changed duration to one minute.

Comment (mwielgus) [reply cluster-autoscaler/core/static_autoscaler_test.go]: Not sure what does it mean exactly but we will not put 1 minute sleep in tests :).

Comment (krzysztof-jastrzebski) [reply cluster-autoscaler/core/static_autoscaler_test.go]: Changed ScaleDownUnreadyTime and ScaleDownUnneededTime  to one minute.

File: cluster-autoscaler/core/static_autoscaler.go [hunk 3]
```diff
@@ -124,7 +124,7 @@ func (a *StaticAutoscaler) RunOnce(currentTime time.Time) errors.AutoscalerError
 	// Update status information when the loop is done (regardless of reason)
 	defer func() {
 		if autoscalingContext.WriteStatusConfigMap {
-			status := a.ClusterStateRegistry.GetStatus(time.Now())
+			status := a.ClusterStateRegistry.GetStatus(currentTime)
```
Comment (mwielgus) [inline cluster-autoscaler/core/static_autoscaler.go]: roll back this change.

**Pull request merged**